crates/ra_mbe/src/lib.rs

```diff
@@ -731,4 +730,95 @@ MACRO_ITEMS@[0; 40)
         }
 "#, r#"extern crate a ; mod b ; mod c {} use d ; const E : i32 = 0 ; static F : i32 = 0 ; impl G {} struct H ; enum I {Foo} trait J {} fn h () {} extern {} type T = u8 ;"#);
     }
+
+    #[test]
+    fn test_block() {
+        let rules = create_rules(
+            r#"
+        macro_rules! foo {
+            ($ i:block) => { fn foo() $ i }
```

Comment:
Huh, I haven't realized that Rust has `block` matcher!

Reply:
The `tricky` one is meta matcher, which match only the content of meta item :)

Reply:
Yeah, I expect we'll hit **a ton** of issues with meta down the road: it can also match doc comments, b/c, in rustc parser, doc comments are a sugar for attributes .

Reply:
🤦